tests/merch.test.ts

```diff
@@ -0,0 +1,100 @@
+import { ForbiddenError } from 'routing-controllers';
+import { UserAccessType } from '../types';
+import { DatabaseConnection, MerchFactory, UserFactory, PortalState } from './data';
+import { ControllerFactory } from './controllers';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('collection editing', () => {
+  test('Appropriate Permission Check for Getting Archived Items', async () => {
+    const conn = await DatabaseConnection.get();
+    const itemOption = MerchFactory.fakeOption();
+    const [item] = MerchFactory.itemsWith({
+      options: [itemOption],
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+      archived: false,
+    });
+
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [user] = UserFactory.with({ accessType: UserAccessType.STANDARD });
+
+    await new PortalState()
+      .createUsers([admin, user])
+      .createMerch([collection])
+      .write();
+
+    const merchStore = ControllerFactory.merchStore(conn);
+
+    const editCollection = {
+      collection: {
+        title: collection.title,
+        description: collection.description,
+        archived: true,
+      },
+    };
```

Comment:
In our API, when you edit objects generally you only need to specify the fields that are changed. This matches the PATCH paradigm of HTTP requests which we predominantly use in our API, as opposed to PUT (you can read about the differences in [this article](https://www.baeldung.com/http-put-patch-difference-spring)). So, simply:
```ts
const collectionEdit = {
  collection: {
    archived: true,
  },
};
```
should suffice.

Reply:
also nit: `editCollection` sounds like a method name since it implies an action is performed (ie. a collection is edited). Variable names should mostly be nouns, so I think `collectionEdit` makes the intention clearer.